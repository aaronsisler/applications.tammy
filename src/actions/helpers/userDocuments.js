export const setUserDocuments = (userDocuments) => ({
    type: 'SET_USER_DOCUMENTS',
    userDocuments,
});

// export const addUserDocument = (userDocument) => ({
//     type: 'ADD_USER_DOCUMENT',
//     userDocument,
// });
