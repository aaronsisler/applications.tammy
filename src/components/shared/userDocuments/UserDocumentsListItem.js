import React from 'react';
import PropTypes from 'prop-types';

export default class UserDocumentsListItem extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <a className="document_list_item" href={`${this.props.downloadURL}`}>
                <div className="document_list_item__title">
                    {this.props.documentName}
                </div>
                <div className="document_list_item__date_uploaded">
                    {this.props.dateUploaded}
                </div>
            </a>
        );
    }
}

UserDocumentsListItem.propTypes = {
    downloadURL: PropTypes.string.isRequired,
    documentName: PropTypes.string.isRequired,
    dateUploaded: PropTypes.string.isRequired,
};
