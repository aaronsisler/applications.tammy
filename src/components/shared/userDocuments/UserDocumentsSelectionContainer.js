import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import UserDocumentsSelectionList from './UserDocumentsSelectionList';

export class UserDocumentsSelectionContainer extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className="user_documents_selection_container">
                <UserDocumentsSelectionList
                    applicationUserDocuments={this.props.applicationUserDocuments}
                    userDocuments={this.props.userDocuments}
                />
            </div>
        );
    }
}

/* istanbul ignore next */
const mapStateToProps = (state) => ({
    applicationUserDocuments: state.application.userDocuments,
    userDocuments: state.userDocuments,
});

export default connect(mapStateToProps)(UserDocumentsSelectionContainer);

UserDocumentsSelectionContainer.propTypes = {
    applicationUserDocuments: PropTypes.array.isRequired,
    userDocuments: PropTypes.array.isRequired,
};
