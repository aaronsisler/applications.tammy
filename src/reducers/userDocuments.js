const userDocumentsReducerDefaultState = [];

const userDocumentsReducer = (state = userDocumentsReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_USER_DOCUMENTS':
            return action.userDocuments;
        default:
            return state;
    }
};

export default userDocumentsReducer;
