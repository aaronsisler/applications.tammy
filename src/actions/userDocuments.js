import database from '../firebase/firebase';
import { setUserDocuments } from 'Actions/helpers/userDocuments';

export const startSetUserDocuments = (userId) => (dispatch) =>
    database.ref(`user_documents/${userId}`).once('value').then((snapshot) => {
        const userDocuments = [];
        snapshot.forEach((childSnapshot) => {
            userDocuments.push({
                id: childSnapshot.key,
                ...childSnapshot.val(),
            });
        });

        dispatch(setUserDocuments(userDocuments));
    })
