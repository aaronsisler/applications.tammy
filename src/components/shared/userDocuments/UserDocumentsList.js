import React from 'react';
import PropTypes from 'prop-types';
import UserDocumentsListItem from './UserDocumentsListItem';

export default class UserDocumentsList extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        if (this.props.userDocuments.length === 0) {
            return (
                <div className="user_documents_list empty">
                    No available items
                </div>
            );
        }

        return (
            <div className="user_documents_list">
                {
                    this.props.userDocuments.map((userDocument, index) =>
                        <UserDocumentsListItem key={index} {...userDocument} />)
                }
            </div>
        );
    }
}

UserDocumentsList.propTypes = {
    userDocuments: PropTypes.array,
};
