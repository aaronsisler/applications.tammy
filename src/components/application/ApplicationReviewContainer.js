import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ApplicationProgressButtonsWidget from './ApplicationProgressButtonsWidget';
import UserDocumentsList from 'Shared/userDocuments/UserDocumentsList';
import UserProfileContainer from 'User/UserProfileContainer';
import { startSubmitApplication } from 'Actions/application';
import { startDecrementCurrentStep, startIncrementCurrentStep } from 'Actions/applicationProcess';

export class ApplicationReviewContainer extends React.Component {
    constructor(props) {
        super(props);
    }

    handleDecrementStep = () => this.props.startDecrementCurrentStep();


    handleIncrementStep = () => {
        this.props.startSubmitApplication(this.props.positionId);
        this.props.startIncrementCurrentStep();
    }

    render() {
        return (
            <div className="application_review_container">
                <div className="application_review_container__wrapper">
                    <ApplicationProgressButtonsWidget
                        handleDecrementStep={this.handleDecrementStep}
                        handleIncrementStep={this.handleIncrementStep}
                    />
                </div>
                <div className="application_review_container__user_documents">
                    <UserDocumentsList userDocuments={this.props.applicationUserDocuments} />
                </div>
                <UserProfileContainer isReadOnly={true} />
            </div>
        );
    }
}

/* istanbul ignore next */
const mapStateToProps = (state) => ({
    applicationUserDocuments: state.application.userDocuments,
});

/* istanbul ignore next */
const mapDispatchToProps = (dispatch) => ({
    startDecrementCurrentStep: () => dispatch(startDecrementCurrentStep()),
    startIncrementCurrentStep: () => dispatch(startIncrementCurrentStep()),
    startSubmitApplication: (positionId) => dispatch(startSubmitApplication(positionId)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ApplicationReviewContainer);

ApplicationReviewContainer.propTypes = {
    applicationUserDocuments: PropTypes.array.isRequired,
    positionId: PropTypes.string.isRequired,
    startDecrementCurrentStep: PropTypes.func.isRequired,
    startIncrementCurrentStep: PropTypes.func.isRequired,
    startSubmitApplication: PropTypes.func.isRequired,
};
