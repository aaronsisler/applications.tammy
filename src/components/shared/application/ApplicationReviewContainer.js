import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ApplicationProgressButtonsWidget from './ApplicationProgressButtonsWidget';
import DocumentList from '../document/DocumentList';
import { startSubmitApplication } from '../../../actions/application';

export class ApplicationReviewContainer extends React.Component {
    constructor(props) {
        super(props);
    }

    handleDecrementCurrentStep = () => {
        this.props.handleDecrementCurrentStep();
    }

    handleIncrementCurrentStep = () => {
        this.props.startSubmitApplication();
        this.props.handleIncrementCurrentStep();
    }

    render() {
        return (
            <div className="application_review_container">
                <ApplicationProgressButtonsWidget
                    currentStep={this.props.currentStep}
                    maxSteps={this.props.maxSteps}
                    handleDecrementCurrentStep={this.handleDecrementCurrentStep}
                    handleIncrementCurrentStep={this.handleIncrementCurrentStep}
                />
                <DocumentList documents={this.props.applicationDocuments} />
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    applicationDocuments: state.application.userDocuments,
})

const mapDispatchToProps = (dispatch) => ({
    startSubmitApplication: () => dispatch(startSubmitApplication()),
})

export default connect(mapStateToProps, mapDispatchToProps)(ApplicationReviewContainer);

ApplicationReviewContainer.propTypes = {
    applicationDocuments: PropTypes.array.isRequired,
    currentStep: PropTypes.number.isRequired,
    maxSteps: PropTypes.number.isRequired,
    handleIncrementCurrentStep: PropTypes.func.isRequired,
    handleDecrementCurrentStep: PropTypes.func.isRequired,
    startSubmitApplication: PropTypes.func.isRequired,
};
