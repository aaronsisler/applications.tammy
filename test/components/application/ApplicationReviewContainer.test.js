import React from 'react';
import { shallow } from 'enzyme';
import { ApplicationReviewContainer } from 'Application/ApplicationReviewContainer';
import userDocuments from '../../fixtures/userDocuments';

describe('ApplicationReviewContainer', () => {
    let wrapper;
    const startClearPosition = jest.fn();
    const startDecrementCurrentStep = jest.fn();
    const startIncrementCurrentStep = jest.fn();
    const startSubmitApplication = jest.fn();

    const buildWrapper = (applicationUserDocuments, userDocumentsInput) => {
        wrapper = shallow(
            <ApplicationReviewContainer
                applicationUserDocuments={applicationUserDocuments}
                startClearPosition={startClearPosition}
                startDecrementCurrentStep={startDecrementCurrentStep}
                startIncrementCurrentStep={startIncrementCurrentStep}
                startSubmitApplication={startSubmitApplication}
                userDocuments={userDocumentsInput}
            />
        );
    };

    beforeEach(() => {
        buildWrapper(userDocuments, userDocuments);
    });

    it('should render ApplicationReviewContainer correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should set the isReadOnly prop to true on UserProfileContainer', () => {
        expect(wrapper.find('Connect(UserProfileContainer)').props().isReadOnly).toBe(true);
    })

    describe('when ApplicationProgressButtonsWidget handleDecrementStep prop is called', () => {
        it('should call startDecrementCurrentStep', () => {
            wrapper.find('Connect(ApplicationProgressButtonsWidget)').prop('handleDecrementStep')();
            expect(startDecrementCurrentStep).toHaveBeenCalled();
        });
    });

    describe('when ApplicationProgressButtonsWidget handleIncrementStep prop is called', () => {
        it('should call startClearPosition', () => {
            wrapper.find('Connect(ApplicationProgressButtonsWidget)').prop('handleIncrementStep')();
            expect(startClearPosition).toHaveBeenCalled();
        });

        it('should call startIncrementCurrentStep', () => {
            wrapper.find('Connect(ApplicationProgressButtonsWidget)').prop('handleIncrementStep')();
            expect(startIncrementCurrentStep).toHaveBeenCalled();
        });

        it('should call startSubmitApplication', () => {
            wrapper.find('Connect(ApplicationProgressButtonsWidget)').prop('handleIncrementStep')();
            expect(startSubmitApplication).toHaveBeenCalled();
        });
    });
});
