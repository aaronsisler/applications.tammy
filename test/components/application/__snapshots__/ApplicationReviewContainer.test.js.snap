// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationReviewContainer should render ApplicationReviewContainer correctly 1`] = `
<div
  className="application_review_container"
>
  <Connect(ApplicationProgressButtonsWidget)
    handleDecrementStep={[Function]}
    handleIncrementStep={[Function]}
  />
  <div
    className="application_review_container__user_documents"
  >
    <UserDocumentsList
      userDocuments={
        Array [
          Object {
            "dateUploaded": "2018/01/01 01:23:45A",
            "documentName": "Document Name 1",
            "downloadUrl": "downloadUrl1",
            "userDocumentId": "firstDocumentId",
          },
          Object {
            "dateUploaded": "2018/02/02 01:23:45A",
            "documentName": "Document Name 2",
            "downloadUrl": "downloadUrl2",
            "userDocumentId": "secondDocumentId",
          },
        ]
      }
    />
  </div>
  <Connect(UserProfileContainer)
    isReadOnly={true}
  />
</div>
`;
